sdk/file_sdk/UnitTests/xmp_ut/xmp_file_tests.cpp

```diff
@@ -14,28 +21,89 @@ class Xmp_Tests : public QObject
   Xmp_Tests();
 
 private Q_SLOTS:
-  void GetTags_FileWithTag_ReturnCorrectTag();
+  void GetTags_FileWithAutomaticTag_ReturnCorrectTag();
+  void GetTags_FileWithManualTag_ReturnCorrectTag();
 };
 
-Xmp_Tests::Xmp_Tests()
-{
+Xmp_Tests::Xmp_Tests() {
 }
 
-void Xmp_Tests::GetTags_FileWithTag_ReturnCorrectTag()
-{
-  /*mip::Tag general( "f42aa342-8706-4288-bd11-ebb85995028c", "General", "", "vakuras@microsoft.com", true, "", "", mip::Method::MANUAL,"72f988bf-86f1-41af-91ab-2d7cd011db47" );
+bool VerifyTags(Tag tag1, Tag tag2) {
+  if (tag1.GetExtendedProperties().size() != tag2.GetExtendedProperties().size())
+    return false;
+
+  for (size_t i = 0; i < tag1.GetExtendedProperties().size(); i++) {
+    if (tag1.GetExtendedProperties()[i].key != tag2.GetExtendedProperties()[i].key ||
+        tag1.GetExtendedProperties()[i].value != tag2.GetExtendedProperties()[i].value ||
+        tag1.GetExtendedProperties()[i].vendor != tag2.GetExtendedProperties()[i].vendor) {
+      return false;
+    }
+  }
+
+  return
+      tag1.GetLabelId() == tag2.GetLabelId() &&
+      tag1.GetLabelName() == tag2.GetLabelName() &&
+      tag1.GetLabelParentId() == tag2.GetLabelParentId() &&
+      tag1.GetEnabled() == tag2.GetEnabled() &&
+      tag1.GetOwner() == tag2.GetOwner() &&
+      tag1.GetSiteId() == tag2.GetSiteId();
+}
+
+void Xmp_Tests::GetTags_FileWithManualTag_ReturnCorrectTag() {
+  mip::Tag general( "f42aa342-8706-4288-bd11-ebb85995028c", "", "", "vakuras@microsoft.com", true, "", mip::Method::MANUAL, "https://rmsibizaapidf.trafficmanager.net/api/72f988bf-86f1-41af-91ab-2d7cd011db47");
+
+  QTemporaryDir tempDir;
+  tempDir.autoRemove();
+  foreach( const QString &fileName, QDir(":xmp_labeled_manual").entryList() )
+  {
+    auto fileNameString = fileName.toStdString();
+    auto extension = fileNameString.substr(fileNameString.find_last_of("."));
+    auto tempFile = tempDir.path() + "/temp" + QString::fromStdString(extension);
+    if (QFile::copy(":xmp_labeled_manual/"+fileName, tempFile)) {

```

Comment:
>if ( [](start = 4, length = 4)

is it necessary?

Reply:
It can fail if something will go wrong.
I will fail the test in that case

---
In reply to: [136713725](https://github.com/AzureAD/rms-sdk-for-cpp/pull/140#discussion_r136713725) [](ancestors = 136713725)